### PR TITLE
Changed kat plugin to parse web search insteed of rss

### PR DESCRIPTION
**Pull request by ZefQ**
seeds and leeches does not work when using kat rss feed.
This commit adds the option to parse the webpage search results insteed.
New config property 'rss' defaults to 'no'

### CODE REVIEW

File: flexget/plugins/search_kat.py [hunk 1]
```diff
@@ -59,7 +68,7 @@ def search(self, task, entry, config):
 
                 log.debug('requesting: %s' % url)
                 try:
-                    r = task.requests.get(url, params=params, raise_status=False)
```
Comment (liiight) [inline flexget/plugins/search_kat.py]: Why did you remove using `task.requests`?

File: flexget/plugins/search_kat.py [hunk 2]
```diff
@@ -69,34 +78,66 @@ def search(self, task, entry, config):
                 elif r.status_code != 200:
                     log.warning('Search returned %s response code' % r.status_code)
                     continue
-                rss = feedparser.parse(r.content)
 
-                ex = rss.get('bozo_exception', False)
-                if ex:
-                    log.warning('Got bozo_exception (bad feed)')
-                    continue
+                entries |= self.parse_rss(r.content) if config.get('rss') else self.parse_web(r.content)
```
Comment (liiight) [inline flexget/plugins/search_kat.py]: that `|=` is a typo, right?

File: flexget/plugins/search_kat.py (hunk 1 above)
Comment (ZefQ) [inline flexget/plugins/search_kat.py]: Yeah what happened to that, probably a error but i don't remember

File: flexget/plugins/search_kat.py (hunk 2 above)
Comment (ZefQ) [inline flexget/plugins/search_kat.py]: No
>>>test = set(['Jane', 'Jack', 'Susan', 'Zack'])
>>>test2 = set(['Jack', 'Sam', 'Susan', 'Janice'])
>>>test |= test2
>>>set(['Zack', 'Sam', 'Susan', 'Jack', 'Jane', 'Janice'])

**Comment by ZefQ** [comment]
Anyhow I think i most take another look at this if it is going to be merged upstream

**Comment by liiight** [comment]
you wanna fix this or should i close?

File: flexget/plugins/search_kat.py [hunk 3]
```diff
@@ -69,34 +78,66 @@ def search(self, task, entry, config):
...
+            entry['url'] = item.enclosures[0]['url']
+            entry['torrent_seeds'] = int(item.torrent_seeds)
+            entry['torrent_leeches'] = int(item.torrent_peers)
+            entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
+            entry['content_size'] = int(item.torrent_contentlength) / 1024 / 1024
+            entry['torrent_info_hash'] = item.torrent_infohash
+
+            entries.add(entry)
 
         return entries
```
Comment (kristenmills) [inline flexget/plugins/search_kat.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/kristen/Flexget/file/2167275111/issues/source?bid=3161667&fileBranchId=3161675#l114)

File: flexget/plugins/search_kat.py [hunk 4]
```diff
@@ -69,34 +78,66 @@ def search(self, task, entry, config):
...
-                if ex:
-                    log.warning('Got bozo_exception (bad feed)')
-                    continue
+                entries |= self.parse_rss(r.content) if config.get('rss') else self.parse_web(r.content)
 
-                for item in rss.entries:
-                    entry = Entry()
-                    entry['title'] = item.title
+        return entries
+
```
Comment (kristenmills) [inline flexget/plugins/search_kat.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/kristen/Flexget/file/2167275111/issues/source?bid=3161667&fileBranchId=3161675#l86)

**Comment by liiight** [comment]
Reopen if relevant
